.github/workflows/07-image-publish.yaml

```diff
@@ -0,0 +1,61 @@
+name: Create and publish Docker image
+
+on:
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}  
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: docker-app
+
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write    
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3  
+
+      - name: Login to Container Reg
+        uses: docker/login-action@v3
+        with: 
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+
+      - name: Build and push image
+        id: push
+        uses: docker/build-push-action@v6
+        with:
+          context: docker-app
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
Something to think about in your future research and experiments: In a world where actions are written by different suppliers and are released with different versions, what do we need to be aware of when sharing data between steps?